src/manifest.c

```diff
@@ -776,24 +779,49 @@ static int write_manifest_plain(struct manifest *manifest)
 	}
 
 	list = g_list_first(manifest->manifests);
+	string_or_die(&manifest_tempdir, "./XXXXXX");
```

Comment:
You mean using the pre-defined STATE_DIR as the template? 'Cause it won't always be /var/lib/update. However, I don't see the point of it being placed in any specific directory, as all of the contents & tempdir are already wiped before update creation ends, so there's no artifacts left over from it.